contracts/oracle/BackedOracle.sol

```diff
@@ -0,0 +1,201 @@
+    }
+
+    function latestRound() external view override returns (uint256) {
+        require(_latestRoundNumber != 0, "No data present");
+
+        return _latestRoundNumber;
+    }
+
+    function latestRoundData()
+        external
+        view
+        override
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        require(_latestRoundNumber != 0, "No data present");
+
+        return (
+            uint80(_latestRoundNumber),
+            _roundData[_latestRoundNumber].answer,
+            _roundData[_latestRoundNumber].timestamp,
+            _roundData[_latestRoundNumber].timestamp,
+            uint80(_latestRoundNumber)
+        );
+    }
+
+    function getAnswer(
+        uint256 roundId
+    ) external view override returns (int256) {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return _roundData[roundId].answer;
+    }
+
+    function getTimestamp(
+        uint256 roundId
+    ) external view override returns (uint256) {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return _roundData[roundId].timestamp;
+    }
+
+    function getRoundData(
+        uint80 roundId
+    )
+        external
+        view
+        override
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        require(roundId <= _latestRoundNumber, "No data present");
+
+        return (
+            roundId,
+            _roundData[roundId].answer,
+            _roundData[roundId].timestamp,
+            _roundData[roundId].timestamp,
+            roundId
+        );
+    }
+
+    function updateAnswer(
+        int192 newAnswer,
+        uint32 newTimestamp
+    ) external onlyRole(UPDATER_ROLE) {
+        int256 latestAnswer = _roundData[_latestRoundNumber].answer;
+        uint256 latestTimestamp = _roundData[_latestRoundNumber].timestamp;
+
+        // Timestamp is actual timestamp
+        require(
+            newTimestamp < block.timestamp,
+            "Timestamp cannot be in the future"
+        );
+
+        // Check that the timestamp is not too old
+        require(
+            block.timestamp - newTimestamp <= MAX_TIMESTAMP_AGE,
+            "Timestamp is too old"
+        );
+
+        // The timestamp is more than the last timestamp
+        require(
+            newTimestamp > latestTimestamp,
+            "Timestamp is older than the last update"
+        );
+
+        // The last update happened more than MIN_UPDATE_INTERVAL ago
+        require(
+            newTimestamp - latestTimestamp > MIN_UPDATE_INTERVAL,
+            "Timestamp cannot be updated too often"
+        );
+
+        // Limit the value to at most MAX_PERCENT_DIFFERENCE% different from the last value
+        if (latestAnswer > 0) {
+            int192 allowedDeviation = int192(
+                (latestAnswer * int8(MAX_PERCENT_DIFFERENCE)) / 100
+            );
+
+            if (newAnswer > latestAnswer + allowedDeviation) {
```

Comment:
Mentioned this before, but just want to make really sure that you know that:

 `updateAnswer(newAnswer)` sometimes will set the answer to a number diffreent from `newAnswer`

This is risky, because no error is raised. For example, lets say the price is `1000`, and the max change is 10%.  Someone wants to set it to 900, but mistakes a decimal and calls `updateAnswer(9000)`. The transaction passes succesfully and the new answer is now `1100`. 

I think the behavior should be to revert if the value is too far off.

Reply:
To put it in other words: price information is way to sensitive to add logic that implicitly converts the answer when setting it

Reply:
Acknowledged. Leaving as is.
The protection is against malicious price manipulation. The updating answer process is automated and so not prone to human "fat-thumb" errors. We are content with this as is.